atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,138 @@
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import shutil
+import tarfile
+import tempfile
+
+from atomic_reactor.constants import (PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY,
+                                      OPERATOR_MANIFESTS_ARCHIVE)
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.util import df_parser, is_scratch_build, get_platforms
+from docker.errors import APIError
+from osbs.utils import Labels
+
+MANIFESTS_DIR_NAME = 'manifests'
+OPERATORS_ARCHIVE_NAME, OPERATORS_ARCHIVE_EXT = OPERATOR_MANIFESTS_ARCHIVE.rsplit('.', 1)
+
+
+class ExportOperatorManifestsPlugin(PostBuildPlugin):
+    """
+    Export operator manifest files
+
+    Fetch and archive operator manifest files from image so they can be
+    uploaded to koji.
+    """
+
+    key = PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, allowed_platform=None, platform=None):
+        super(ExportOperatorManifestsPlugin, self).__init__(tasker, workflow)
+
+        self.allowed_platform = allowed_platform
+        if platform:
+            self.platform = platform
+        else:
+            self.platform = os.uname()[4]
+
+    def has_operator_manifest(self):
+        """
+        Check if Dockerfile sets the operator manifest label
+
+        :return: bool
+        """
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = Labels(dockerfile.labels)
+        try:
+            _, operator_label = labels.get_name_and_value(Labels.LABEL_TYPE_OPERATOR_MANIFESTS)
+        except KeyError:
+            operator_label = 'false'
+        return operator_label == "true"
+
+    def is_orchestrator(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        if self.allowed_platform != self.platform:
+            self.log.info("Only platform [%s] will upload operators metadata. Skipping",
+                          self.allowed_platform)
+            return False
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+        if not self.should_run():
+            return
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Do we care about removing this afterwards?

Reply:
Should we? I did not chose the best name for that directory (changed).

The final operator manifests zip file will be saved in this directory. If we want to remove it, then I understand we want to remove the manifest zip file after it is uploaded, is that correct?

Reply:
Oh, I see. I guess that's fine.